polyfill/lib/ecmascript.mjs

```diff
@@ -1558,35 +1426,15 @@ export function InterpretISODateTimeOffset(
     if (MathAbs(ISODateToEpochDays(balanced.year, balanced.month - 1, balanced.day)) > 1e8) {
       throw new RangeErrorCtor('date/time outside of supported range');
     }
-    const epochNs = GetUTCEpochNanoseconds(
-      balanced.year,
-      balanced.month,
-      balanced.day,
-      balanced.hour,
-      balanced.minute,
-      balanced.second,
-      balanced.millisecond,
-      balanced.microsecond,
-      balanced.nanosecond
-    );
+    const epochNs = GetUTCEpochNanoseconds(balanced);
     ValidateEpochNanoseconds(epochNs);
     return epochNs;
   }
 
-  if (MathAbs(ISODateToEpochDays(year, month - 1, day)) > 1e8) {
+  if (MathAbs(ISODateToEpochDays(isoDate.year, isoDate.month - 1, isoDate.day)) > 1e8) {
```

Comment:
ISODateToEpochDays is supposed to be a 1-to-1 replacement for ECMA-262 [MakeDay](https://tc39.es/ecma262/#sec-makeday) but using mathematical values, because some of the ECMA-262 operations related to MakeDay don't correctly handle the edges of the time value range. There is an [issue](https://github.com/tc39/ecma262/issues/1087) open for ECMA-262 to fix this, and when it's fixed the intention is to go back to using MakeDay here.

MakeDay and the other existing ECMA-262 ops use 0-based month numbers, so ISODateToEpochDays does as well. That's why we have the decrement. We can maybe improve this down the line when Temporal reaches stage 4.

About the "abs(ISODateToEpochDays(year, month - 1, day)) > 1e8" pattern, it's there because of #2729 and #2985. I like the idea of abstracting it, especially because it can disappear when after that ECMA-262 issue is fixed.